### PR TITLE
Support aria-roledescription.

### PR DESCRIPTION
**Pull request by michaelDCurran**
### Link to issue number:
None.

### Summary of the issue:
In ARIA 1.1, authors can override the way a control's role is spoken/brailled by providing the aria-roledescription attribute.
An example of this might be calling a toggle button a "light switch" as  in a particular situation that is perhaps how it was visually styled.

### Description of how this pull request fixes the issue:
NVDA has already supported roleText internally which NVDAObject subclasses could override (e.g. Powerpoint shapes).  This PR  makes use of roleText to expose aria-roledescription from IA2 and MSHTML web browsers (including Firefox, Chrome, and Internet Explorer).
This PR also ensures that if roleText is provided for a field in browseMode that would normally get a presentation category of layout (and therefore  the role is silenced) that  the field ends up being classed as either container or singleline depending on whether it is block or not. Thus, custom roleText will be presented appropriately.

### Testing performed:
Testcase: http://www.ets-research.org/ia11ylab/aria11spec/
Tested in Firefox, Chrome and Internet Explorer.

### Known issues with pull request:
Edge does not support this attribute yet.

### Change log entry:
What's new:
* custom roles via the aria-role-description attribute are now supported in Firefox, Chrome and Internet Explorer.

### CODE REVIEW

File: source/speech.py [hunk 1]
```diff
@@ -1130,7 +1130,9 @@ def getControlFieldSpeech(attrs,ancestorAttrs,fieldType,formatConfig=None,extraD
 	else:
 		tableID = None
 
-	roleText=getSpeechTextForProperties(reason=reason,role=role)
+	roleText=attrs.get('roleText')
```
Comment (derekriemer) [inline source/speech.py]: What if we changed this to this?
roleText=attrs.get('roleText', lambda:getSpeechTextForProperties(reason=reason,role=role))
That will prevent the function from ever needing called in the roletext case, and removes that if.

**Review by feerrenrut — APPROVED** [review]
Could you please update the description to specify whether this works with Chrome?

**Comment by michaelDCurran** [comment]
aria-roledescription works with Chrome. I have updated the description accordingly.

**Comment by derekriemer** [comment]
Why does this say "region" then "slide"?

**Pull request merged**